### ISSUE TITLE
FeathersMenuItem does not properly show keyboard focus

### ISSUE THREAD

**Issue by SpecificProtagonist** [bevyengine/bevy#24933]
bevy_feathers shows keyboard focus via a feathers.focus.ring outline, like so:

<img width="524" height="111" alt="Image" src="https://github.com/user-attachments/assets/442baa9b-6b45-46c8-bbb4-b8cd541058ec" />

`FeathersMenuItem` fails to do so, either not showing the focus at all (in the `feathers_gallery` example, click on "Menu" and then press the down key):

<img width="154" height="179" alt="Image" src="https://github.com/user-attachments/assets/defdcc43-cec3-4596-aa79-860f1c31765b" />

or showing it by changing the background color (select the menu with tab, then press the down key) (in the screenshot, the second item is also highlighted as due to mouse hover):

<img width="155" height="176" alt="Image" src="https://github.com/user-attachments/assets/b374bbc6-2fae-4fba-bbc7-03361eee7bc4" />

Edit: Eager menus currently don't work due to a seperate bug (#24935), but the issue is visible on lazy menus.

**Comment by cold-leopard** [comment]
Fixing the second scenario here is pretty straightforward, adding `FocusIndicator` to the `FeathersMenuItem` scene gets the proper focus indicator to be used.

The first scenario seems trickier as need to make the focus indicator visible first, and not sure the best way to do that


 https://github.com/bevyengine/bevy/blob/96e3bcfd87f4cb6372dd9da8b5318f3e64899a01/crates/bevy_ui_widgets/src/list.rs#L190-L195 the listbox widget toggles visibility of the focus indicator on key events (which I think is why the list and select examples in the demo work). Something similar could be done here with the `menu_on_key_event`, but wondering if this should just be pulled into a more general arrow navigation module similar to https://github.com/bevyengine/bevy/blob/96e3bcfd87f4cb6372dd9da8b5318f3e64899a01/crates/bevy_input_focus/src/tab_navigation.rs
